src/core/component.js

```diff
@@ -34,6 +34,7 @@ var Component = module.exports.Component = function (el, attrValue, id) {
   this.el = el;
   this.id = id;
   this.attrName = this.name + (id ? '__' + id : '');
+  this.el.components[this.attrName] = this;
```

Comment:
If you call `setAttribute` in the init routine the new call will trigger a new component instantiation nested in the first one because everything happens in the constructor. Setting the component on the entity early in the constructor prevents the nested initializations when calling set attribute.